### DISCUSSION TITLE
Why are some Rxn Delta G = 1e7?

### DISCUSSION THREAD

**Discussion by jspmccain** [SysBioChalmers/yeast-GEM#351] [General]
Hi!

I'm just wondering why some of the deltaG values are equal to 1e7, for example r_0362 or r_0439.

Thanks!

**Comment by edkerk** [comment]
@cheng-yu-zhang Can you answer this?

**Reply by cheng-yu-zhang** [reply]
The reactions with 1e7 are caused by 53 metabolites containing groups with unknown energy in their composition. In the yetfl, they manually set `deltaG=1e7`. If we apply thermodynamic constraints, better to avoid them or compute the deltaG using other methods.
53 metabolites are listed as follows.

Mets
--
4-hydroxybenzoyl-CoA
4-methyl-5-(2-phosphonooxyethyl)thiazole
5-(2-hydroxyethyl)-4-methylthiazole
6,7-dimethyl-8-(1-D-ribityl)lumazine
coproporphyrinogen III
FAD
ferroheme b
FMN
heme a
heme o
hydrogen sulfide
L-proline
N-acetylputrescine
palmitoyl-CoA
potassium
precorrin-2
protoporphyrin
protoporphyrinogen
riboflavin
siroheme
sirohydrochlorin
sodium
TDP
tetracosanoyl-CoA
thiamine
TMP
thiamine(1+) triphosphate(4-)
UDP-N-acetyl-alpha-D-glucosamine
uroporphyrinogen III
3-oxoicosanoyl-CoA
3-hydroxyicosanoyl-CoA
trans-2,cis-9-octadecadienoyl-CoA
(R)-3-hydroxyicosanoyl-CoA
cadmium(2+)
Mn(2+)
L-methionine (S)-S-oxide
Zn(2+)
L-Methionine S-oxide
4-nitrophenyl phosphate
4-nitrophenol
Ca(2+)
ADP-5-ethyl-4-methylthiazole-2-carboxylate
Ethylnitronate
4-Acetamidobutanoate
Mg(2+)
polyphosphate
Cu2(+)
porphyrin
hypotaurine
N-phosphocreatine
N(omega)-phospho-L-arginine
triphosphate
3-sulfino-L-alanine

**Reply by jspmccain** [reply]
Great, thank you! I'm also wondering about the code that generates the delta G's in the spreadsheet, is that currently available also?

**Reply by cheng-yu-zhang** [reply]
Sorry, a matlab function computing delta G's is not available. But you can try the python code `../yeast-GEM/deltaG/code/run_dgpredictor.py` in pull request [`feat: add metDeltaG and rxnDeltaG fields #330`](https://github.com/SysBioChalmers/yeast-GEM/pull/330) to compute the reaction delta G's.
